### ISSUE TITLE
Remove project.json support

### ISSUE THREAD

**Issue by nkolev92** [NuGet/Home#7199]
Self descriptive. 
I propose we remove project.json support in 5.0 and dev16. 

The only project.json native customer was UWP and they have since moved on to PackageReference. 

Now might be a perfect time to remove PJ and cleanup the APIs that are brought along with that. 

[2347704](https://dev.azure.com/devdiv/DevDiv/_workitems/edit/2347704)

**Comment by MikeRosoft** [comment]
Technically, project.json was used in ASP.NET Core projects in Visual Studio 2015. In Visual Studio 2017 this was also replaced with ProjectReference.

**Comment by nkolev92** [comment]
This can't happen in 5.0/16.0 due to some dependencies that are still using project.json. 

@anangaur @rrelyea 
Anything special we need to do ensure that we can remove support in 17.0? 
Some deprecation announcement?

**Comment by anangaur** [comment]
We should announce on the announcements repo may be. Not sure if we should do anything more than that. But the announcement could be when we start previews of 17.0. Doing it now won’t serve much.

**Comment by nkolev92** [comment]
Makes sense, figured to get clarity on this early enough, before it become an afterthought as we focus on the current release :)

**Comment by JonDouglas** [comment]
Okay, audible plan is d17 & 6.0 :)

**Comment by nkolev92** [comment]
https://github.com/dotnet/docs/issues/47968

Notification sent.
